framework/include/problems/FEProblemBase.h

```diff
@@ -1900,6 +1900,15 @@ class FEProblemBase : public SubProblem, public Restartable
     _n_max_nl_pingpong = n_max_nl_pingpong;
   }
 
+  /// method setting the minimum number of nonlinear iterations before performing divergence checks
+  void setNLForcedIterations(const unsigned int nl_forced_its) { _nl_forced_its = nl_forced_its; }
```

Comment:
Could we consider just force the number of iterations regardless the algorithm convergence unless we hit a fundamental issue such as "NAN"?

Some users like to force a certain number of nonlinear iterations even though the algorithm could converge before that

Reply:
> Could we consider just force the number of iterations regardless the algorithm convergence unless we hit a fundamental issue such as "NAN"?
>
> Some users like to force a certain number of nonlinear iterations even though the algorithm could converge before that


I think we can just need to add the condition to the converged tests

Reply:
Great